### PR TITLE
fix(auth): oidc authentication when running multiple instances of the agent

### PR DESCRIPTION
**Pull request by DrRaider**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: src/services/oidc-client-manager.js [hunk 1]
```diff
@@ -37,23 +37,44 @@ class OidcClientManagerService {
...
-            token_endpoint_auth_method: 'none',
-            redirect_uris: [callbackUrl],
-          },
-          error,
+      let registrationPromise;
+      if (this.env.FOREST_CLIENT_ID) {
+        registrationPromise = new issuer.Client({
+          client_id: this.env.FOREST_CLIENT_ID,
+          redirect_uris: [callbackUrl],
+          token_endpoint_auth_method: 'none',
```
Comment (VincentMolinie) [inline src/services/oidc-client-manager.js:45]: This is duplicate with the line below

File: src/services/oidc-client-manager.js [hunk 2]
```diff
@@ -37,23 +37,44 @@ class OidcClientManagerService {
...
+          this.cache.delete(callbackUrl);
+          throw error;
         });
-        this.cache.delete(callbackUrl);
-        throw error;
-      });
+      } else {
+        registrationPromise = issuer.Client.register({
+          token_endpoint_auth_method: 'none',
+          redirect_uris: [callbackUrl],
```
Comment (VincentMolinie) [inline src/services/oidc-client-manager.js:62]: This is duplicated too

File: src/services/oidc-client-manager.js [hunk 3]
```diff
@@ -37,23 +37,44 @@ class OidcClientManagerService {
...
-        this.logger.error('Unable to register the client', {
-          configuration,
-          registration: {
-            token_endpoint_auth_method: 'none',
-            redirect_uris: [callbackUrl],
-          },
-          error,
+      let registrationPromise;
+      if (this.env.FOREST_CLIENT_ID) {
+        registrationPromise = new issuer.Client({
```
Comment (VincentMolinie) [inline src/services/oidc-client-manager.js:42]: Actually you should just initalialise an object registration

Comment (VincentMolinie) [reply src/services/oidc-client-manager.js:42]: Because all the rest is just duplicated

File: test/services/oidc-client-manager.test.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const OidcClientManagerService = require('../../src/services/oidc-client-manager');
 
 describe('service > OidcClientManager', () => {
-  function setupTest() {
+  function setupTest(withClientId) {
```
Comment (VincentMolinie) [inline test/services/oidc-client-manager.test.js:4]:
```suggestion
  function setupTest(withClientId = false) {
```

**Review by LouisClisson — APPROVED** [review]
### ✅  &nbsp;Code review
👍
### ✅  ⚠️&nbsp;Automatic tests
👍
### ✅  ⚠️&nbsp;Manual tests
I tested the authentication flow and the current changes does not change the behavior. The fix should allow us to use oidc with agent running on multiple instance
### ✅  ⚠️&nbsp;PR title
👍

**Pull request merged**